### PR TITLE
Add Keyboard application and move character mapping to userland

### PR DESCRIPTION
**Pull request by asliturk**
This is the first step of moving keyboard character mapping logic from kernel to user space. This PR contains new `Keyboard` application for settings, `WindowServer` enhancments and `LibCore` objects for mapping.

Next step is `VirtualConsole` and `keymap` application. And then remove mapping code from `KeyboardDriver` class.

### CODE REVIEW

**Comment by awesomekling** [comment]
Hmm.. what is the benefit of doing mapping in userspace if you still have to map in `VirtualConsole` anyway?

**Comment by asliturk** [comment]
User can change it as a normal user. And we can persist this settings in config file (every time i have to change it before user terminal). 

I am working on `VirtualConsole` but it is broken for now. My design for 'VirtualConsole' prefers to initilize it with the US layout with hard coded `CharacterMapData` (maybe FS is not ready at this point). superuser use `keymap` cmd utility to change it.

And of course simplify `KeyboardDriver` implementation.

File: Servers/WindowServer/Screen.cpp [hunk 1]
```diff
@@ -162,8 +163,20 @@ void Screen::on_receive_mouse_data(const MousePacket& packet)
...
+
+    auto user_land_karakter = m_character_map.get_char(kernel_event);
+
+    auto message = make<KeyEvent>(kernel_event.is_press() ? Event::KeyDown : Event::KeyUp, kernel_event.key, user_land_karakter, kernel_event.modifiers());
     Core::EventLoop::current().post_event(WindowManager::the(), move(message));
 }
 
+void Screen::set_keyboard_map(String keymap_file_name)
+{
+    m_character_map = LibCore::CharacterMap(keymap_file_name);
```
[the remark this answers is not in the document]
Comment (asliturk) [reply Servers/WindowServer/Screen.cpp:175]: All new variable and class names changed to character_map. This PR is about character mapping not key mapping. I am working on other change set to provide a keymap (keyboard layout) change. This two mapping must be separeted. In Turkish Q layout scan code `'0x2E'` is `'C'` and `Key_C`, in Turkish F layout `'0x2E'` is mapped to `'V'` char. On Turkish F keyboard this scan code (0x2E) must be mapped to `Key_V` not `Key_C`.

File: Libraries/LibCore/CharacterMap.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <AK/String.h>
+#include <Kernel/KeyCode.h>
+#include <LibCore/CharacterMapFile.h>
+
+namespace LibCore {
+
+class CharacterMap {
+
```
[the remark this answers is not in the document]
Comment (asliturk) [reply Libraries/LibCore/CharacterMap.h:10]: LibKeyboard :)

**Comment by asliturk** [comment]
I temporarily closing this PR. After Ramadan i will be back :)

**Pull request closed without merging**